### PR TITLE
RavenDB-17624 allow to open session in subscription batch only once

### PR DESCRIPTION
**Pull request by shaharhikri**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-17624

### Additional description

Fixi - Creating and disposing of session per subscription batch item leads to "blittable object has been disposed" exception.

### Type of change

- Bug fix

### How risky is the change?

- Low 

### Backward compatibility

- Not relevant

### Is it platform specific issue?

- No

### Documentation update

- No documentation update is needed 

### Testing 

- Tests have been added that prove the fix is effective or that the feature works

### Is there any existing behavior change of other features due to this change?

- No

### UI work

- No UI work is needed

### CODE REVIEW

File: src/Raven.Client/Documents/Subscriptions/SubscriptionBatch.cs [hunk 1]
```diff
@@ -111,6 +112,12 @@ public IAsyncDocumentSession OpenAsyncSession(SessionOptions options)
 
         private IAsyncDocumentSession OpenAsyncSessionInternal(SessionOptions options)
         {
+            if (_sessionOpened)
+            {
+                throw new InvalidOperationException("'SubscriptionBatch' can open only 1 session");
```
Comment (ppekrol) [inline src/Raven.Client/Documents/Subscriptions/SubscriptionBatch.cs]: Let's be human: `Session can only be opened once per each Subscription batch.`

File: test/SlowTests/Issues/RavenDB-17624.cs [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+                pVal1 = c1.ProcessedOn;
+                pVal2 = c2.ProcessedOn;
+            }
+
+            Assert.Equal(last, pVal1);
+            Assert.Equal(null, pVal2);
+        }
+
+        public class Command
```
Comment (ppekrol) [inline test/SlowTests/Issues/RavenDB-17624.cs]: private, there are pre-made classes in the project

File: test/SlowTests/Issues/RavenDB-17624.cs [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+namespace SlowTests.Issues
+{
+    public class RavenDB_17624 : RavenTestBase
+    {
+        public RavenDB_17624(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        [Fact]
+        public async Task ForbidOpeningMoreThenOneSessionPerSubscriptionBatch()
```
Comment (ppekrol) [inline test/SlowTests/Issues/RavenDB-17624.cs:26]: I do not understand why this test is so complex, has some many console writelines, etc

**Review by ppekrol — CHANGES_REQUESTED** [review]

File: src/Raven.Client/Documents/Subscriptions/SubscriptionBatch.cs [hunk 4]
```diff
@@ -111,6 +112,12 @@ public IAsyncDocumentSession OpenAsyncSession(SessionOptions options)
 
         private IAsyncDocumentSession OpenAsyncSessionInternal(SessionOptions options)
         {
+            if (_sessionOpened)
```
Comment (garayx) [inline src/Raven.Client/Documents/Subscriptions/SubscriptionBatch.cs:120]: we should add this check to `OpenSessionInternal()` as well,  no?

File: test/SlowTests/Issues/RavenDB-17624.cs [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+        {
+            var store = GetDocumentStore();
```
Comment (garayx) [inline test/SlowTests/Issues/RavenDB-17624.cs]: using

File: test/SlowTests/Issues/RavenDB-17624.cs (hunk 3 above)
Comment (shaharhikri) [reply test/SlowTests/Issues/RavenDB-17624.cs:26]: It was originally copied from the original scenario of the user,
Look at the previous commits.
I'll delete the unnecessary code from this test.

File: test/SlowTests/Issues/RavenDB-17624.cs [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+            var cts = new CancellationTokenSource();
+            var last = DateTime.UtcNow;
+            InvalidOperationException exception = null;
+            var t = worker.Run(async batch =>
+            {
+                using (var session = batch.OpenAsyncSession())
+                {
+                }
+
+                exception = await Assert.ThrowsAsync<InvalidOperationException>(async () =>
```
Comment (karmeli87) [inline test/SlowTests/Issues/RavenDB-17624.cs]: please fix the warning

**Pull request merged**